### PR TITLE
Add ApplyGuardrailScope for security guardrail evaluation tracing

### PR DESCRIPTION
**Pull request by nikhilNava**
## Summary

Introduces a new OpenTelemetry tracing scope (\ApplyGuardrailScope\) that captures security guardrail evaluations as spans. This enables observability into content safety, policy enforcement, and risk assessment decisions made during agent operations.

Port of [microsoft/opentelemetry-distro-dotnet#109](https://github.com/microsoft/opentelemetry-distro-dotnet/pull/109) (excluding ETW support, which is not applicable to the Python SDK).

### New Contracts
- **\GuardrailDetails\** — Frozen dataclass capturing guardian evaluation metadata: target type, decision, provider info, policy details, and content hash for forensic correlation.
- **\GuardrailFinding\** — Represents an individual risk finding with severity, category, and confidence score.
- **\GuardrailDecisionType\** — Enum for well-known guardian decisions (Allow, Audit, Deny, Modify, Warn).
- **\GuardrailRiskSeverity\** — String constants for risk severity levels (None, Low, Medium, High, Critical).
- **\GuardrailTargetType\** — Constants for guardrail targets (LlmInput, LlmOutput, ToolCall, etc.).

### New Tracing Scope
- **\ApplyGuardrailScope\** — Context manager scope following the same pattern as \InvokeAgentScope\, \InferenceScope\, and \ExecuteToolScope\. Supports recording findings post-creation via \
ecord_finding()\, updating decisions via \
ecord_decision()\, and recording content output via \
ecord_content_output()\.

### OpenTelemetry Constants
- Added \microsoft.security.*\ and \microsoft.guardian.*\ attribute keys to constants.
- Added \pply_guardrail\ to the exporter operation name filter set.

### Tests
- 14 unit tests covering scope creation, finding recording, attribute emission, and edge cases.

## Design Notes

- Guardian spans are children of the operation they protect (e.g., inference or tool execution spans) via \SpanKind.INTERNAL\.
- Multiple guardian spans may exist under a single operation when guardrails are chained.
- Follows the existing Core + Extensions pattern and context manager scope lifecycle.
- All 250 existing observability core tests continue to pass.

### CODE REVIEW

File: libraries/microsoft-agents-a365-observability-core/microsoft_agents_a365/observability/core/apply_guardrail_scope.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        if details.content_modified is not None:
+            self.set_tag_maybe(SECURITY_CONTENT_MODIFIED_KEY, details.content_modified)
+
+        # Correlation
+        self.set_tag_maybe(SECURITY_EXTERNAL_EVENT_ID_KEY, details.external_event_id)
+
+        # Request context
+        if request is not None:
+            self.set_tag_maybe(SECURITY_CONTENT_INPUT_VALUE_KEY, request.content)
+            self.set_tag_maybe(GEN_AI_CONVERSATION_ID_KEY, request.conversation_id)
```
[the remark this answers is not in the document]
Comment (nikhilNava) [reply libraries/microsoft-agents-a365-observability-core/microsoft_agents_a365/observability/core/apply_guardrail_scope.py:152]: conversation id is not PII

File: libraries/microsoft-agents-a365-observability-core/microsoft_agents_a365/observability/core/apply_guardrail_scope.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        self.set_tag_maybe(SECURITY_CONTENT_INPUT_HASH_KEY, details.content_input_hash)
+        if details.content_modified is not None:
+            self.set_tag_maybe(SECURITY_CONTENT_MODIFIED_KEY, details.content_modified)
+
+        # Correlation
+        self.set_tag_maybe(SECURITY_EXTERNAL_EVENT_ID_KEY, details.external_event_id)
+
+        # Request context
+        if request is not None:
+            self.set_tag_maybe(SECURITY_CONTENT_INPUT_VALUE_KEY, request.content)
```
[the remark this answers is not in the document]
Comment (nikhilNava) [reply libraries/microsoft-agents-a365-observability-core/microsoft_agents_a365/observability/core/apply_guardrail_scope.py]: fixed

**Review by juliomenendez — APPROVED** [review]

**Review by gwharris7 — APPROVED** [review]
